src/py/mat3ra/made/tools/convert.py

```diff
@@ -0,0 +1,174 @@
+        ],
+        "units": "crystal",
+        "cell": __round__(structure.lattice.matrix.tolist()),
+        "constraints": [],
+    }
+
+    lattice = {
+        "a": __round__(structure.lattice.a),
+        "b": __round__(structure.lattice.b),
+        "c": __round__(structure.lattice.c),
+        "alpha": __round__(structure.lattice.alpha),
+        "beta": __round__(structure.lattice.beta),
+        "gamma": __round__(structure.lattice.gamma),
+        "units": {"length": "angstrom", "angle": "degree"},
+        "type": "TRI",
+        "vectors": {
+            "a": __round__(structure.lattice.matrix[0].tolist()),
+            "b": __round__(structure.lattice.matrix[1].tolist()),
+            "c": __round__(structure.lattice.matrix[2].tolist()),
+            "alat": 1,
+            "units": "angstrom",
+        },
+    }
+
+    material_data = {
+        "name": structure.formula,
+        "basis": basis,
+        "lattice": lattice,
+        "isNonPeriodic": not structure.is_ordered,
+        "_id": "",
+        "metadata": {"boundaryConditions": {"type": "pbc", "offset": 0}},
+        "isUpdated": True,
+    }
+
+    return material_data
+
+
+def to_poscar(material_or_material_data: Union[Material, Dict[str, Any]]) -> str:
+    """
+    Converts material object in ESSE format to a POSCAR string.
+
+    Args:
+        material_or_material_data (dict|class): A dictionary containing the material information in ESSE format.
+
+    Returns:
+        str: A POSCAR string.
+    """
+    structure = to_pymatgen(material_or_material_data)
+    poscar = Poscar(structure)
+    return poscar.get_str()
+
+
+def from_poscar(poscar: str) -> Dict[str, Any]:
+    """
+    Converts a POSCAR string to a material object in ESSE format.
+
+    Args:
+        poscar (str): A POSCAR string.
+
+    Returns:
+        dict: A dictionary containing the material information in ESSE format.
+    """
+    structure = Structure.from_str(poscar, "poscar")
+    return from_pymatgen(structure)
+
+
+def to_ase(material_or_material_data: Union[Material, Dict[str, Any]]) -> Atoms:
+    """
+    Converts material object in ESSE format to an ASE Atoms object.
+
+    Args:
+        material_or_material_data (dict|class): A dictionary containing the material information in ESSE format.
+
+    Returns:
+        Any: An ASE Atoms object.
+    """
+    structure = to_pymatgen(material_or_material_data)
+    return AseAtomsAdaptor.get_atoms(structure)
+
+
+def from_ase(ase_atoms: Atoms) -> Dict[str, Any]:
+    """
+    Converts an ASE Atoms object to a material object in ESSE format.
+
+    Args:
+        ase_atoms (Atoms): An ASE Atoms object.
+
+    Returns:
+        dict: A dictionary containing the material information in ESSE format.
+    """
+    structure = AseAtomsAdaptor.get_structure(ase_atoms)
+    return from_pymatgen(structure)
+
+
+def convert_material_args_kwargs_to_atoms(func: Callable) -> Callable:
```

Comment:
I like this approach to generalize functions to take arguments in any format

Reply:
Yes, the idea is that all tools should be applicable to Material class instances and the conversion should be done in the decorator layer, if needed.